### ISSUE TITLE
onErrorRetry not triggered after tab blur+focus with useSWRImmutable

### ISSUE THREAD

**Issue by BertrandBordage** [vercel/swr#2543]
# Bug report

## Description / Observed Behavior

`useSWRImmutable` stops retrying errored requests if you switch to a different browser tab then go back to the application.
This happens using the default `onErrorRetry` as well as a custom one.

## Expected Behavior

Like `useSWR`, `useSWRImmutable` should keep retrying until the request ends successfully.
Of course, once the first successful request is finished `useSWRImmutable` should never run again, unlike `useSWR`.

## Repro Steps / Code Example

https://codesandbox.io/s/useswrimmutable-onerrorretry-bug-89n0de

## Additional Context

SWR version 2.1.2

**Comment by milhamm** [comment]
This also happens using `useSWR` when `revalidateOnFocus` is set to `false`

**Comment by Shubhdeep12** [comment]
I encountered this today. 

Tried to resolve this issue in swrimmutable.


I am pretty sure there is some issue in middleware an in the way we are using mergeobjects helper func as even if i set the  values of revalidate keys to true in index.ts of swrimmutable, even then also the keys are coming as false in final config. And it's not revalidating on focus.


I'll debug more😅

**Comment by Shubhdeep12** [comment]
I encountered this.

I tried to debug it. 

When it comes to softrevalidate on focus and reconnecting. we check these conditions 

on focus - `if (  getConfig().revalidateOnFocus &&   now > nextFocusRevalidatedAt &&  isActive()  )`
and on reconnect -  `if (  getConfig().revalidateOnReconnect &&  isActive()  )`

But, in the case of swrimmutable we don't want to revalidate the data but what if API fails and we have to run errorOnRetry. then we have to revalidate and there comes executing softRevalidate.

I think we should also add a condition checking `getConfig().shouldRetryOnError`

any thoughts @promer94
